### PR TITLE
MINOR: Fix typo and refactor new group coordinator tests

### PR DESCRIPTION
**Pull request by squah-confluent**
* Fix a small typo.
* Merge the tests for fetchOffsets and fetchAllOffsets together into
  parameterized tests since they share the same structure.
* Use Topic.GROUP_METADATA_TOPIC_NAME instead of __consumer_offsets in
  new group coordinator tests.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java [hunk 1]
```diff
@@ -1140,20 +1154,22 @@ public void testFetchOffsets(
 
         if (requireStable) {
             when(runtime.scheduleWriteOperation(
-                ArgumentMatchers.eq("fetch-offsets"),
+                ArgumentMatchers.eq(fetchAllOffsets ? "fetch-all-offsets" : "fetch-offsets"),
                 ArgumentMatchers.eq(new TopicPartition("__consumer_offsets", 0)),
```
Comment (m1a2st) [inline group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java]: I think `__consumer_offsets` can replace by `Topic.GROUP_METADATA_TOPIC_NAME`, use constant is better than string

File: group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java [hunk 2]
```diff
@@ -1140,20 +1154,22 @@ public void testFetchOffsets(
...
                 ArgumentMatchers.eq(Duration.ofMillis(5000)),
                 ArgumentMatchers.any()
             )).thenReturn(CompletableFuture.completedFuture(response));
         } else {
             when(runtime.scheduleReadOperation(
-                ArgumentMatchers.eq("fetch-offsets"),
+                ArgumentMatchers.eq(fetchAllOffsets ? "fetch-all-offsets" : "fetch-offsets"),
                 ArgumentMatchers.eq(new TopicPartition("__consumer_offsets", 0)),
```
Comment (m1a2st) [inline group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java]: ditto

File: group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java [hunk 3]
```diff
@@ -1222,152 +1255,24 @@ public void testFetchOffsetsWithWrappedError(
...
-        service.startup(() -> 1);
-
-        OffsetFetchRequestData.OffsetFetchRequestGroup request =
-            new OffsetFetchRequestData.OffsetFetchRequestGroup()
-                .setGroupId("group");
-
         when(runtime.scheduleWriteOperation(
-            ArgumentMatchers.eq("fetch-all-offsets"),
+            ArgumentMatchers.eq(fetchAllOffsets ? "fetch-all-offsets" : "fetch-offsets"),
             ArgumentMatchers.eq(new TopicPartition("__consumer_offsets", 0)),
```
Comment (m1a2st) [inline group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java]: ditto

**Review by m1a2st — COMMENTED** [review]
Hello @squah-confluent, I left some comments, PTAL. Thanks

File: group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java (hunk 1 above)
Comment (squah-confluent) [reply group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorServiceTest.java]: `__consumer_offsets` is used throughout the new group coordinator tests. I think it makes sense to update all instances in the new group coordinator at once and pushed an update for it.

**Review by m1a2st — APPROVED** [review]
LGTM

**Review by dajac — APPROVED** [review]
LGTM

**Pull request merged**